### PR TITLE
set min PL as 1.7

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Reaction to #910

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [ ] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by otaj — APPROVED** [review]

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
         # Something is wrong with windows tests, will need to debug separately.
         os: [ubuntu-20.04, macOS-11]
         # todo: seems there are some issues with py3.10 while install dependencies
-        python-version: ["3.7", "3.10"]
+        python-version: ["3.7", "3.9"]
```
Comment (otaj) [inline .github/workflows/ci-tests.yml:23]: we still want to install 3.10, the issue is that 3.10 doesn't have all of the oldest things (such as torchvision)

**Review by otaj — APPROVED** [review]

File: .github/workflows/ci-tests.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-tests.yml:23]: just fixed in [f624673](https://github.com/Lightning-AI/lightning-bolts/pull/930/commits/f624673bb9aa1010ddbfe3301746abfe83892034)

Comment (otaj) [reply .github/workflows/ci-tests.yml]: in here :) 80be648

**Comment by Borda** [comment]
@otaj, the py3.10 is failing with TV, see:
https://github.com/Lightning-AI/lightning-bolts/actions/runs/3376424612/jobs/5604130908

**Comment by Borda** [comment]
Seems there is some issue with py3.10 and `onnx`

**Comment by otaj** [comment]
I'm on it

**Comment by otaj** [comment]
sparseml is forcing old onnx version, which would require compilation from source on 3.10 in the CI. We won't be doing that, so for now CI runs on 3.7 and 3.9 :sadpanda:

**Pull request merged**